hack/update-deps.sh

```diff
@@ -13,16 +13,15 @@ set -o pipefail
 cd "${ROOT_DIR}"
 
 # This controls the knative release version we track.
-KN_VERSION="release-1.4"
-EVENTING_VERSION="release-v1.4"
+KN_VERSION="release-1.5"
+EVENTING_VERSION="release-v1.5"
 EVENTING_KAFKA_VERSION="release-v1.4"
```

Comment:
@pierDipi do we have a plan to remove the deprecated eventing_kafka bits. Should I go on and do it in another PR?

Reply:
I think we can remove the code dependency now, thanks for your help on this!

Reply:
We should still keep the test-source-image reference, only the code/vendor dependency can be removed for now.

Reply:
We use the dep in tests too: https://github.com/openshift-knative/serverless-operator/blob/main/test/upgrade/upgrade_test.go#L27, so let's remove that in another PR.

Reply:
that code dependency is transitively imported by eventing-kafka-broker, so we will keep it but it's not gonna be enforced anymore

Reply:
I can remove the one line if that is the goal here. It has no effect, it does not touch go.mod. Is that what you are describing?